agent/agent_test.go

```diff
@@ -500,6 +500,59 @@ func TestAgent_AddService(t *testing.T) {
 	}
 }
 
+func TestAgent_AddServices_AliasUpdateCheckNotReverted(t *testing.T) {
+	t.Parallel()
+	a := NewTestAgent(t, t.Name(), `
+		node_name = "node1"
+	`)
+	defer a.Shutdown()
+
+	// It's tricky to get an UpdateCheck call to be timed properly so it lands
+	// right in the middle of an addServiceInternal call so we cheat a bit and
+	// rely upon alias checks to do that work for us.  We add enough services
+	// that probabilistically one of them is going to end up properly in the
+	// critical section.
```

Comment:
Did this test fail reliably without the fix? Might be nice to mention how reliable it was given that it's probabilistic.

Reply:
Surprisingly it reliably failed every time with the first number I picked (10). I'll amend the comment.